hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -172,7 +172,7 @@ private void unschedule() {
         // So the outputBuffer is empty, so we are going to unschedule ourselves.
         scheduled.set(false);
 
-        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty()) {
+        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty() && currentPacket == null) {
```

Comment:
A concern about our new WriteHandler. Old way of handling read and write was there since very early days of hazelcast and it is tested and proven over these years that it is stable. And as far as I know, we changed it so that it becomes more readable. I personally found two major bugs in new one. And there is another critical bug found by @jerrinot . My question is that are we confident enough release with this one? I might be missing something. Does this change has any other proven advantages  like performance other than readability? If there is performance improvement, It make sense to insist on continue with new WriteHandler implementation. Note: I did not change the client side WrieHandler to new one, because I think it is not mature enough.